check_pedigree.py

```diff
@@ -0,0 +1,1217 @@
+    duos = [list(x) for x in duos]
+
+    inferred_duos = duos
+    given_duos = ped_duos
+                    
+    # find differences between given and inferred trios
+    confirmed_duos = []
+    inferred_only_duos = []
+    given_only_duos = []
+                              
+    for duo in inferred_duos:
+        if duo in given_duos:
+            confirmed_duos.append(duo)
+        else:
+            inferred_only_duos.append(duo)
+
+    for duo in given_duos:
+        if duo not in inferred_duos:
+            given_only_duos.append(duo)
+            duo_out = re.sub("'", "", str(duo)) 
+            out_review.write("given_only_duo\t{duo_out}\t.\n".format(**locals()))
+
+
+    duos_accounted = inferred_only_duos + confirmed_duos
+
+    hl_duos_accounted  = hl.literal(duos_accounted) if duos_accounted else hl.empty_array(hl.tarray(hl.tstr))
+    hl_inferred_only_duos  = hl.literal(inferred_only_duos) if inferred_only_duos else hl.empty_array(hl.tarray(hl.tstr))
+    hl_confirmed_duos  = hl.literal(confirmed_duos) if confirmed_duos else hl.empty_array(hl.tarray(hl.tstr))
+
+    # remove duos from temp kin ht
+    ht = ht.filter(hl_duos_accounted.contains(ht.pair), keep=False)
+
+    # annotate master kin with dup annotations
+    master_kin_ht = master_kin_ht.annotate(call = hl.if_else(hl_duos_accounted.contains(master_kin_ht.pair), "parent_offspring", master_kin_ht.call))
+    master_kin_ht = master_kin_ht.annotate(final_evaluation = hl.if_else(hl_inferred_only_duos.contains(master_kin_ht.pair), "inferred_only_duo", master_kin_ht.final_evaluation))
+    master_kin_ht = master_kin_ht.annotate(final_evaluation = hl.if_else(hl_confirmed_duos.contains(master_kin_ht.pair), "confirmed_duo", master_kin_ht.final_evaluation))
+
+    out_summary.write("Number of confirmed duos: " + str(len(confirmed_duos)) + '\n')
+    out_summary.write("Number of inferred only duos: " + str(len(inferred_only_duos)) + '\n')
+    out_summary.write("Number of given only duos: " + str(len(given_only_duos)) + '\n\n')
+    
+    return(ht, master_kin_ht, duos, given_only_duos)
+
+
+
+def evaluate_siblings(
+    ht: hl.Table,
+    master_kin_ht: hl.Table,
+    out_summary: io.TextIOWrapper
+) -> Tuple[hl.Table, hl.Table, hl.expr.ArrayExpression]:
+
+    """
+    Determine if siblings are confirmed or inferred only, annotate master kinship table with sibling decisions
+    :param ht: hl.Table
+    :param master_ht: hl.Table
+    :return: ht with sibling samples removed and master kinship ht with annotated sibling decisions, sibling pairs
+    :rtype: Tuple[hl.Table, hl.Table, hl.expr.ArrayExpression]
+    """
+    
+    sibling_ht = ht.filter(ht.relationship == SIBLINGS)
+    
+    # confirmed sibs are those where the final decision (parents) matched
+    # need to check they are not both none/no decision because that's not really a match
+    confirmed_sibs = sibling_ht.filter((sibling_ht.decision_i == sibling_ht.decision_j) & (sibling_ht.decision_i != ["no_decision"]))
```

Comment:
Both confirmed and unconfirmed sibs can be collected together. Moreover, you only use them within hail, so no need to get them into python first:
```
_sibs = sibling_ht.aggregate(
    hl.agg.group_by(
        (sibling_ht.decision_i == sibling_ht.decision_j) & (sibling_ht.decision_i != ["no_decision"]), 
        hl.agg.collect(sibling_ht.pair)
    ),
    _localize=False
)
```